packages/valory/skills/market_creation_manager_abci/behaviours.py

```diff
@@ -2291,6 +2291,54 @@ def _get_balance(self, account: str) -> Generator[None, None, Optional[int]]:
         balance = cast(int, ledger_api_response.state.body.get("get_balance_result"))
         return balance
 
+    def _eligible_questions_to_answer(
+        self, unanswered_questions: List[Dict[str, Any]]
+    ) -> List[str]:
+        """Determine the eligible questions to answer at this time"""
+        now = self.last_synced_timestamp
+        eligible_questions_id = []
+
+        self.context.logger.info(
+            f"Answer retry intervals: {self.params.answer_retry_intervals}"
+        )
+
+        for question in unanswered_questions:
+            question_id = question["question"]["id"].lower()
+
+            if question_id in self.shared_state.questions_responded:
+                continue
+
+            if question_id not in self.shared_state.questions_requested_mech:
+                self.shared_state.questions_requested_mech[question_id] = {}
+                self.shared_state.questions_requested_mech[question_id][
+                    "question"
+                ] = question
+                self.shared_state.questions_requested_mech[question_id]["retries"] = []
+
+            retries = self.shared_state.questions_requested_mech[question_id]["retries"]
+            n_retries = len(retries)
+            time_since_last_retry = now - retries[-1] if retries else 0
+            retry_period = self.params.answer_retry_intervals[
+                min(n_retries, len(self.params.answer_retry_intervals) - 1)
+            ]
+            if n_retries == 0 or time_since_last_retry > retry_period:
+                eligible_questions_id.append(question_id)
+
+        self.context.logger.info(
+            f"Determined {len(eligible_questions_id)} eligible questions to answer."
+        )
+
+        num_questions = min(
+            len(eligible_questions_id), self.params.multisend_batch_size
+        )
+        random.seed(self.last_synced_timestamp)
+        random_questions_id = random.sample(eligible_questions_id, num_questions)
```

Comment:
Randomness should always come from an external source.

Reply:
It comes from the synchronized timestamp of all the agents.

Reply:
The collect randomness is after this state, so this need refactor. Opened an issue for this: https://github.com/valory-xyz/market-creator/issues/103